modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs

```diff
@@ -170,7 +174,8 @@ INamedTypeSymbol symbol
                     .Select(s => s?.Initializer ?? null)
                     .FirstOrDefault();
 
-                string? value = initializer?.Value.ToString();
+                // Fully qualify the value to avoid issues with namespaces.
+                var value = initializer?.Value?.FullQualifiedSyntax(sm);
```

Comment:
Following my other review comment, the semantic model could be obtained here:

```suggestion
                string? value;

                if (initializer != null)
                {
                    var sm = context.Compilation.GetSemanticModel(initializer.SyntaxTree);

                    // Fully qualify the value to avoid issues with namespaces.
                    value = initializer.Value.FullQualifiedSyntax(sm);
                }
                else
                {
                    value = null;
                }
```

It can be extracted to a method if it's too verbose.

Reply:
I don't know how much it costs to get the semantic model. If it ends up being a problem, a way to reduce repeated calls could be to iterate by syntax trees of the class symbol locations rather than by properties. In any case, I wouldn't worry about that in this PR.